File: python3/Sort and Search/dutch_national_flag.py
from typing import List


def dutch_national_flag(nums: List[int]) -> None:
    i, left, right = 0, 0, len(nums) - 1
    while i <= right:
        # Swap 0s with the element at the left pointer.
        if nums[i] == 0:
            nums[i], nums[left] = nums[left], nums[i]
            left += 1
            i += 1
        # Swap 2s with the element at the right pointer.
        elif nums[i] == 2:
            nums[i], nums[right] = nums[right], nums[i]
            right -= 1
        else:
            i += 1

File: cpp/Sort and Search/dutch_national_flag.cpp
#include <vector>
#include <algorithm>

void dutchNationalFlag(std::vector<int>& nums) {
    int i = 0, left = 0, right = nums.size() - 1;
    while (i <= right) {
        // Swap 0s with the element at the left pointer.
        if (nums[i] == 0) {
            std::swap(nums[i], nums[left]);
            left++;
            i++;
        }
        // Swap 2s with the element at the right pointer.
        else if (nums[i] == 2) {
            std::swap(nums[i], nums[right]);
            right--;
        }
        else {
            i++;
        }
    }
}